instrumentation/rails/lib/opentelemetry/instrumentation/rails/patches/action_controller/metal.rb

```diff
@@ -14,6 +14,10 @@ module Metal
             def dispatch(name, request, response)
               rack_span = OpenTelemetry::Instrumentation::Rack.current_span
               rack_span.name = "#{self.class.name}##{name}" if rack_span.context.valid? && !request.env['action_dispatch.exception']
```

Comment:
The reason we opted for this approach is that we believe using the controller name and action as the span name is more conventional in the context of a rails application.

As you pointed out this is not following the spec, would it make more sense for your use case if we added a configuration that allowed you to set the span name to the path template instead?

Reply:
More or less, the spec is a little confused about that.

About the [name](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/api.md#span)
> The span name concisely identifies the work represented by the Span, for example, an RPC method name, a function name, or the name of a subtask or stage within a larger computation. The span name SHOULD be the most general string that identifies a (statistically) interesting class of Spans, rather than individual Span instances while still being human-readable. That is, "get_user" is a reasonable name, while "get_user/314159", where "314159" is a user ID, is not a good name due to its high cardinality. Generality SHOULD be prioritized over human-readability.

So in that case make sense to me, in a rails app, it behavior you used.
But we have a conflict when we start to talk about the [HTTP Server semantic conventions](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/http.md#http-server-semantic-conventions)

> Given an inbound request for a route (e.g. "/users/:userID?") the name attribute of the span SHOULD be set to this route. If the route does not include the application root, it SHOULD be prepended to the span name.
>
> If the route cannot be determined, the name attribute MUST be set as defined in the general semantic conventions for HTTP.

In that case, the span name AND route should be the same? That doesn't make sense to me (could be a problem in specification not here).

The problem (to me) in only change the span name is the current behavior in [Datadog exporter](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/exporter/datadogexporter/translate_traces.go#L441) they use the route to build the resource (and show grouped in UI). I only thought of adding a route here, but maybe we need to change the exporter behavior to look to other attributes.

Reply:
Hey, @ericmustin I see you have some commits in Datadog exporter, could you help us here in that discussion?
What do you think is the better approach?